### PR TITLE
fix: converge interleaved audio samples

### PR DESCRIPTION
**Pull request by EzraEllette**
## description
I identified an issue with the collection of audio data. devices with 2 channels use an interleaved format of [left,right,left,right] and this wasn't accounted for. By converging the left and right audio samples, the transcripts are working again.

related issue: #521
related issue: #525

## type of change
- [x] bug fix
- [ ] new feature
- [ ] breaking change
- [ ] documentation update

## how to test
1. on mac use display audio as an input
2. test transcription

### CODE REVIEW

File: screenpipe-audio/src/audio_processing.rs [hunk 1]
```diff
@@ -78,3 +78,19 @@ pub fn average_noise_spectrum(audio: &[f32]) -> f32 {
 
     total_sum / audio.len() as f32
 }
+
+pub fn stereo_to_mono(audio: &[f32], channels: u16) -> Vec<f32> {
```
Comment (louis030195) [inline screenpipe-audio/src/audio_processing.rs:82]: @EzraEllette what happen if device has more than 2 channels like 3-4 on windows?

Comment (EzraEllette) [reply screenpipe-audio/src/audio_processing.rs:82]: Good call

File: screenpipe-audio/src/stt.rs [hunk 2]
```diff
@@ -315,12 +315,7 @@ pub async fn stt(
         audio_input.data.as_ref().to_vec()
     };
 
-    let audio_data = if audio_input.device.device_type == DeviceType::Input {
-        normalize_v2(&audio_data)
-    } else {
-        // ! for some reason buggy on output devices
-        audio_data
-    };
+    let audio_data = normalize_v2(&audio_data);
```
Comment (louis030195) [inline screenpipe-audio/src/stt.rs:318]: https://github.com/thewh1teagle/vibe/blob/276a6a20b711ecb6c1aa080d1906ab83269626a0/core/src/audio.rs#L54

just referencing if useful

File: screenpipe-audio/src/audio_processing.rs (hunk 1 above)
Comment (EzraEllette) [reply screenpipe-audio/src/audio_processing.rs]: @louis030195 I updated this.

**Comment by EzraEllette** [comment]
@louis030195 did you notice a difference with this after more testing? I noticed a big difference in transcription quality for stereo audio streams.

**Pull request merged**